integration_test/cases/interval.exs

```diff
@@ -297,4 +298,39 @@ defmodule Ecto.Integration.IntervalTest do
     assert [^datetime] =
            TestRepo.all(from p in User, select: datetime_add(p.inserted_at, ^dec, "year"))
   end
+
+  test "datetime_add with naive_datetime_usec" do
+    TestRepo.insert!(%Article{published_at: ~N[2014-01-01 02:00:00.000001]})
+    dec = Decimal.new(1500)
+
+    assert [~N[2014-01-01 02:00:00.001501]] =
```

Comment:
Please add a clause where the value we are calling `datetime_add` is an interpolated value and not a database column, something like:

```elixir
  assert [~N[2014-01-01 02:00:00.001501]] =
    TestRepo.all(from p in Article, select: datetime_add(^~N[2014-01-01 02:00:00.000001], 1500, "microsecond"))
```

and the same to the test below. I think it won't work because we are casting external values to naive_datetime, which means loss of precision. If so, you will have to change `:naive_datetime` in the snippet below to `:naive_datetime_usec`:

https://github.com/elixir-ecto/ecto/blob/295482fe5323614ae71078e93929657c7538bf71/lib/ecto/query/builder.ex#L120-L121